app/controllers/vocabulary_controller.php

```diff
@@ -63,14 +63,27 @@ public function of($username, $lang = null)
     {
         $this->helpers[] = 'Pagination';
         $this->helpers[] = 'CommonModules';
+
+        $username = Sanitize::paranoid($username, array('_'));
         
         $userId = $this->User->getIdFromUsername($username);
+
+        if (!$userId) {
+            $this->Session->setFlash(
+                __('No user with this username: ', true).$username
```

Comment:
I'm taking care of this. There are a few other places where a placeholder should be used.